packages/model-viewer/src/three-components/PlacementBox.ts

```diff
@@ -0,0 +1,131 @@
+/* @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BufferGeometry, DoubleSide, Float32BufferAttribute, Material, Mesh, MeshBasicMaterial, PlaneBufferGeometry, Vector2, Vector3} from 'three';
+
+import {Damper} from './Damper';
+import Model from './Model';
+import {ModelScene} from './ModelScene';
+
+const RADIUS = 0.2;
+const LINE_WIDTH = 0.03;
+const MAX_OPACITY = 0.75;
+const SEGMENTS = 12;
+const DELTA_PHI = Math.PI / (2 * SEGMENTS);
+
+const vector2 = new Vector2();
+
+const addCorner =
```

Comment:
A comment would not hurt for this one (feel free to make it a private class member if you really don't want to document it).

Reply:
Yeah, I couldn't make it a member because it gets called before `super`. I'm attempting to keep it private by not exporting it, but a doc is still a good idea.

Reply:
The joys of inheriting from Three.js classes...